### PR TITLE
Unit- & Integration-Testing

### PR DESCRIPTION
**Pull request by GolexPrime**
This PR introduces the means of conducting unit- and integration tests.
For this purpose [Jest](https://jestjs.io/) is being utilized in combination with [React Native Testing Library](https://testing-library.com/docs/react-native-testing-library/intro/).

In short, this PR:
- configures the jest & babel setup
- integrates a mock-renderer
- integrates a mock-redux-store
- updates the documentation
- adds several example tests

### CODE REVIEW

File: src/screens/checkIn/checkInContainer.js [hunk 1]
```diff
@@ -151,32 +152,42 @@ class CheckInContainer extends Component {
 	 * tries to procure the current questionnaire
 	 */
 	getQuestionnaire = async () => {
+
+		let lala
```
Comment (mbastian93) [inline src/screens/checkIn/checkInContainer.js:156]: undescriptive variable name

File: src/screens/login/loginContainer.js [hunk 2]
```diff
@@ -55,7 +55,8 @@ class LoginContainer extends Component {
 	 * after the component mounted (to clean the state). also triggers the auto-login if 
 	 * configured in appConfig.js
 	 */
-	componentDidMount = () => {
+	componentDidMount = async () => {
```
Comment (mbastian93) [inline src/screens/login/loginContainer.js:58]: why did you change it to 'async'?

**Review by mbastian93 — COMMENTED** [review]
looks good in general

some files miss a new line and the end of file

File: src/screens/checkIn/checkInContainer.js (hunk 1 above)
Comment (GolexPrime) [reply src/screens/checkIn/checkInContainer.js]: yep, missed that one.

File: __mocks__/mocks.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            start_date: "2021-07-15T04:00:00.000Z",
+            study_id: "7bfc3b07-a97d-4e11-8ac6-b970c1745476",
```
[the remark this answers is not in the document]
Comment (GolexPrime) [reply __mocks__/mocks.js]: no, a remnant from an earlier state.

File: src/screens/login/loginContainer.js (hunk 2 above)
Comment (GolexPrime) [reply src/screens/login/loginContainer.js]: at one point I had problems getting the timing right for the automated login (should it be activated and running in development mode). so i tried to run the event handler asynchronously, which actually worked out - but the solution provided by React Native Testing Library (using the `waitFor()` function) is in my eyes much better as the code we want to test does not need to be changed in any way for that to work out.

so short answer: missed to undo it

File: __tests__/CheckIn.test.js [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+    start_date: "2021-07-15T04:00:00.000Z",
+    study_id: "7bfc3b07-a97d-4e11-8ac6-b970c1745476",
```
[the remark this answers is not in the document]
Comment (GolexPrime) [reply __tests__/CheckIn.test.js]: ah, copy & paste... my old nemesis! we meet again...

**Comment by GolexPrime** [comment]
@mbastian93 addressed your concerns

**Review by mbastian93 — APPROVED** [review]
